server/server.go

```diff
@@ -47,6 +69,81 @@ func (cs *cribbageServer) Serve() {
 	}
 }
 
+func (cs *cribbageServer) ginPostRegister(c *gin.Context) {
+	u, p, err := decodeUserAndPass(c)
+	if err != nil {
+		c.String(http.StatusInternalServerError, `Error: %s`, err)
+		return
+	}
+	creds, err := auth.NewCredentials(u, p)
+	if err != nil {
+		c.String(http.StatusInternalServerError, `Error: %s`, err)
+		return
+	}
+	player := model.Player{
+		Credentials: creds,
+		ID:          model.PlayerID(creds.Username),
+		Name:        `player`,
+	}
+	err = createPlayer(context.Background(), player)
+	if err != nil {
+		c.String(http.StatusInternalServerError, `Error: %s`, err)
+		return
+	}
+	// TODO make the key more secret (environment var), change expiration duration to much shorter
+	jwtSvc := auth.NewJWTService(`somethingSecret`, time.Minute*180)
+	cookie, err := jwtSvc.CreateCookie(creds.Username)
+	if err != nil {
+		c.String(http.StatusInternalServerError, `Error: %s`, err)
+		return
+	}
+	c.SetCookie(cookie.Name, cookie.Value, cookie.MaxAge, cookie.Path, cookie.Domain, cookie.Secure, cookie.HttpOnly)
+}
+
+func (cs *cribbageServer) ginPostLogin(c *gin.Context) {
+	u, p, err := decodeUserAndPass(c)
+	if err != nil {
+		c.String(http.StatusInternalServerError, `Error: %s`, err)
+		return
+	}
+	player, err := getPlayer(model.PlayerID(u))
+	if err != nil {
+		if err == persistence.ErrPlayerNotFound {
+			c.String(http.StatusUnauthorized, `Invalid credentials`)
+			return
+		}
+		c.String(http.StatusInternalServerError, `Error: %s`, err)
+	}
+	isAuthorized, err := auth.ValidateCredentials(u, p, player.Credentials)
+	if err != nil {
+		c.String(http.StatusInternalServerError, `Error: %s`, err)
+		return
+	}
+	if !isAuthorized {
+		c.String(http.StatusUnauthorized, `Invalid credentials`)
+		return
+	}
+	jwtSvc := auth.NewJWTService(`somethingSecret`, time.Minute*180)
+	cookie, err := jwtSvc.CreateCookie(player.Credentials.Username)
+	if err != nil {
+		c.String(http.StatusInternalServerError, `Error: %s`, err)
+		return
+	}
+	c.SetCookie(cookie.Name, cookie.Value, cookie.MaxAge, cookie.Path, cookie.Domain, cookie.Secure, cookie.HttpOnly)
+}
+
+func decodeUserAndPass(c *gin.Context) (string, string, error) {
+	var creds struct {
+		Username string `json:"username"`
+		Password string `json:"password"`
+	}
+	err := json.NewDecoder(c.Request.Body).Decode(&creds)
```

Comment:
If I'm reading this correctly, does this mean that we're emitting the password in plaintext (AKA json) over the network?

Reply:
We're receiving the password in the body of the request (not emitting it)

Reply:
-- what I'm asking that if the traffic isn't encrypted (which I don't think we enforce) and somebody is sniffing traffic, won't they be able to read the password out of the http packet?

Reply:
Yarp. I spose we'd need to use https to make this secure